tasks/setup_debian.yml

```diff
@@ -10,26 +10,63 @@
         state: present
         update_cache: true
 
-    - name: Install mariadb repository key
+    - name: Ensure keyrings dir exists
+      ansible.builtin.file:
+        path: /etc/apt/keyrings
+        state: directory
+        owner: root
+        group: root
+        mode: "0755"
+
+    - name: Fetch mariadb repository key
       ansible.builtin.get_url:
-        url: "https://mariadb.org/mariadb_release_signing_key.asc"
-        dest: "/etc/apt/trusted.gpg.d/mariadb.asc"
+        url: https://mariadb.org/mariadb_release_signing_key.asc
+        dest: /etc/apt/keyrings/mariadb-archive-keyring.asc
         # //TEMP todo
         # checksum: sha256:http://example.com/path/sha256sum.txt
         owner: root
         group: root
         mode: "0644"
+      register: fetch_repository_key
+
+    - name: Dearmor mariadb repository key
+      when: fetch_repository_key is changed
+      ansible.builtin.command: >
+        gpg /etc/apt/keyrings/mariadb-archive-keyring.asc
+        --dearmor
+        --output /etc/apt/keyrings/mariadb-archive-keyring.gpg
```

Comment:
This command and the one above could be simplified to:

```yml
- ansible.builtin.shell: >
    curl -sSL https://mariadb.org/mariadb_release_signing_key.asc
    | gpg --dearmor --output /etc/apt/keyrings/mariadb-archive-keyring.gpg
```
But:
- remove the temporary file at /etc/apt/keyrings/mariadb-archive-keyring.asc 
- raises some linting errors
- can't easily check if the key changed or not